### DISCUSSION TITLE
[BUG?] the REST API that get workflow run returns wrong pull request info

### DISCUSSION THREAD

**Discussion by Laurenfrost** [community/community#24674] [Apps, API and Webhooks]
<p>Apologize for adding that question mark. Actually I’m not really sure if it a bug, but when I tried to get the pull request info which triggered the workflow run, I found a different pull request.</p>
<p>In my repo <code>Laurenfrost/ActionsTest</code>, I closed the pull_request number 5, and that trigger a workflow run <a href="https://github.com/Laurenfrost/ActionsTest/actions/runs/1578145440" rel="noopener nofollow ugc">https://github.com/Laurenfrost/ActionsTest/actions/runs/1578145440</a>.</p>
<p>However, when I GET the workflow run info with <a href="https://api.github.com/repos/Laurenfrost/ActionsTest/actions/runs/1578145440" rel="noopener nofollow ugc">https://api.github.com/repos/Laurenfrost/ActionsTest/actions/runs/1578145440</a>, the <code>pull_requests</code> showed it is number 6. That confused me a lot.</p>
<p>Does anyone know how does it happen?</p>
<pre><code class="lang-json">{
  "id": 1578145440,
  "name": "Show Trigger Events",
  "node_id": "WFR_kwLOErVgq85eEJag",
  "head_branch": "next",
  "head_sha": "09e5238cadb749785f29da4c562857a90ac55dce",
  "run_number": 32,
  "event": "pull_request",
  "status": "completed",
  "conclusion": "success",
  "workflow_id": 16281477,
  "check_suite_id": 4652771218,
  "check_suite_node_id": "CS_kwDOErVgq88AAAABFVOnkg",
  "url": "https://api.github.com/repos/Laurenfrost/ActionsTest/actions/runs/1578145440",
  "html_url": "https://github.com/Laurenfrost/ActionsTest/actions/runs/1578145440",
  "pull_requests": [
    {
      "url": "https://api.github.com/repos/Laurenfrost/ActionsTest/pulls/6",
      "id": 804305832,
      "number": 6,
      "head": {
        "ref": "next",
        "sha": "64a34d649db20f60693909135135b3291f638f5b",
        "repo": {
          "id": 313876651,
          "url": "https://api.github.com/repos/Laurenfrost/ActionsTest",
          "name": "ActionsTest"
        }
      },
      "base": {
        "ref": "new",
        "sha": "462e752da3c5c29945b40cce9583875397565045",
        "repo": {
          "id": 313876651,
          "url": "https://api.github.com/repos/Laurenfrost/ActionsTest",
          "name": "ActionsTest"
        }
      }
    }
  ],
  "created_at": "2021-12-14T13:53:34Z",
  "updated_at": "2021-12-14T13:53:48Z",
  "run_attempt": 1,
  "run_started_at": "2021-12-14T13:53:34Z",
  "jobs_url": "https://api.github.com/repos/Laurenfrost/ActionsTest/actions/runs/1578145440/jobs",
  "logs_url": "https://api.github.com/repos/Laurenfrost/ActionsTest/actions/runs/1578145440/logs",
  "check_suite_url": "https://api.github.com/repos/Laurenfrost/ActionsTest/check-suites/4652771218",
  "artifacts_url": "https://api.github.com/repos/Laurenfrost/ActionsTest/actions/runs/1578145440/artifacts",
  "cancel_url": "https://api.github.com/repos/Laurenfrost/ActionsTest/actions/runs/1578145440/cancel",
  "rerun_url": "https://api.github.com/repos/Laurenfrost/ActionsTest/actions/runs/1578145440/rerun",
  "previous_attempt_url": null,
  "workflow_url": "https://api.github.com/repos/Laurenfrost/ActionsTest/actions/workflows/16281477",
  "head_commit": {
    "id": "09e5238cadb749785f29da4c562857a90ac55dce",
    "tree_id": "7f8ed692c1fb874e98ae16d6311279636c57d6aa",
    "message": "Delete TEST-com.bmwcarit.barefoot.analysis.DBCANTest.xml",
    "timestamp": "2021-12-14T13:37:17Z",
    "author": {
      "name": "Linwei Ye",
      "email": "bellfahle@outlook.com"
    },
    "committer": {
      "name": "GitHub",
      "email": "noreply@github.com"
    }
  }
...
}
</code></pre>

**Comment by jsoref** [comment, marked as answer]
<p>I’m pretty sure <code>pull_requests</code> is the list of open pull requests, it’s provided for convenience and not particularly related to the event.</p>

**Comment by Laurenfrost** [comment]
<p>Thank you! That makes sense.</p>
<p>It seems the document not mention the usage of <code>pull_requests</code> in the response. Could you please tell me where shoud I go to know about that.</p>

**Comment by jsoref** [comment]
<p>The “documentation” is here:</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/rest/reference/actions#get-a-workflow-run">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/rest/reference/actions#get-a-workflow-run" target="_blank" rel="noopener nofollow ugc">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/2119212/181085991-3a95bbd2-d554-4a78-a112-c98b68dda453.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/rest/reference/actions#get-a-workflow-run" target="_blank" rel="noopener nofollow ugc">Actions - GitHub Docs</a></h3>



  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>If you’re up for it, you could file a bug:</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/github/docs/issues/">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/github/docs/issues/" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/2119212/181085992-23e45e6a-7b2c-4073-91e3-fdcd9762016c.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/0/2/02826fcda65c44b75fb3103690764d09c041f860_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/3X/0/2/02826fcda65c44b75fb3103690764d09c041f860_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/0/2/02826fcda65c44b75fb3103690764d09c041f860.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/0/2/02826fcda65c44b75fb3103690764d09c041f860_2_10x10.png"></div>

<h3><a href="https://github.com/github/docs/issues/" target="_blank" rel="noopener nofollow ugc">Issues · github/docs</a></h3>

  <p>The open-source repo for docs.github.com. Contribute to github/docs development by creating an account on GitHub.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>Personally, I agree that the documentation is inadequate, it has a field, but in the example there are 0 entries, so it isn’t possible to divine what it is. (If it had 3, one a couple before the PR and one after, and one somewhere, that’d make it fairly clear that it isn’t related to the run…)</p>

**Comment by Laurenfrost** [comment]
<p>Couldn’t agree more. Maybe I should open an isssue.</p>
<p>Thank you every much!</p>
